### PR TITLE
fix(docs): document module generation

### LINKED ISSUE #718: Update documentation for building modules
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When we expanded module support beyond the Terraform registry (and for submodules), the documentation wasn't really updated to account for that. Both https://github.com/hashicorp/terraform-cdk/blob/main/docs/working-with-cdk-for-terraform/using-providers-and-modules.md and https://github.com/hashicorp/terraform-cdk/blob/main/docs/working-with-cdk-for-terraform/cdktf-json.md don't really explain the new functionality.

### References
#584 
#563

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
Closes #718

### CODE REVIEW

**Comment by jsteinich** [comment]
Might want to mention the expanded configuration options in https://github.com/hashicorp/terraform-cdk/blob/main/docs/working-with-cdk-for-terraform/cdktf-json.md. Specifically how source and name can be specified.

**Comment by DanielMSchmidt** [comment]
@jsteinich Oh yeah sounds like a good idea :)

File: docs/working-with-cdk-for-terraform/using-modules.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Using Modules
```
Comment (ansgarm) [inline docs/working-with-cdk-for-terraform/using-modules.md:1]: Maybe name it `Terraform Modules` in the heading to make clear that this is not about the "modules" one would use for organizing a CDKTF project (which would rather make use of custom constructs) instead.

File: docs/working-with-cdk-for-terraform/using-modules.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+## Installing Modules
+
+For using modules on the terraform registry, see [cdktf.json](./cdktf-json.md).
```
Comment (ansgarm) [inline docs/working-with-cdk-for-terraform/using-modules.md:5]: *Terraform Registry

File: docs/working-with-cdk-for-terraform/using-modules.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+For using modules from other sources (local, github, etc), you have two options:
```
Comment (ansgarm) [inline docs/working-with-cdk-for-terraform/using-modules.md:6]: *Github

File: docs/working-with-cdk-for-terraform/cdktf-json.md [hunk 4]
```diff
@@ -39,7 +46,20 @@ Community providers have to provide a fully qualified name, e.g. to define the D
 
 #### Modules
 
-Similar to 3rd Party providers, the full registry namespace should be provided. Please note that only modules from the registry are supported at this point.
+Similar to 3rd Party providers, the full registry namespace should be provided. For local modules please use the object format:
```
Comment (ansgarm) [inline docs/working-with-cdk-for-terraform/cdktf-json.md:49]: not your edit but: *party

File: docs/working-with-cdk-for-terraform/cdktf-json.md [hunk 5]
```diff
@@ -13,13 +13,20 @@ export enum Language {
   GO = "go",
 }
 
+export interface TerraformDependencyConstraint {
```
Comment (ansgarm) [inline docs/working-with-cdk-for-terraform/cdktf-json.md:16]: Nice 👍

File: docs/getting-started/typescript.md [hunk 6]
```diff
@@ -117,7 +117,7 @@ You specify it in your local `cdktf.json` file, like this:
 ```
 
 Next you need to run `cdktf get` which generates provider-specific bindings.
-To save some time you can also use pre-built provider, please visit the ["Using Providers and Modules"-Guide](../working-with-cdk-for-terraform/using-providers-and-modules.md) for more on that topic.
+To save some time you can also use pre-built provider, please visit the ["Using Providers"-Guide](../working-with-cdk-for-terraform/using-providers.md) for more on that topic.
```
Comment (ansgarm) [inline docs/getting-started/typescript.md:120]: *providers

File: docs/getting-started/python.md [hunk 7]
```diff
@@ -141,7 +141,7 @@ You specify it in your local `cdktf.json` file, like this:
 ```
 
 Next you need to run `cdktf get` which generates provider-specific bindings.
-To save some time you can also use pre-built provider, please visit the ["Using Providers and Modules"-Guide](../working-with-cdk-for-terraform/using-providers-and-modules.md) for more on that topic.
+To save some time you can also use pre-built provider, please visit the ["Using Providers"-Guide](../working-with-cdk-for-terraform/using-providers.md) for more on that topic.
```
Comment (ansgarm) [inline docs/getting-started/python.md:144]: *providers

File: docs/getting-started/java.md [hunk 8]
```diff
@@ -153,7 +153,7 @@ You specify it in your local `cdktf.json` file, like this:
 ```
 
 Next you need to run `cdktf get` which generates provider-specific bindings.
-To save some time you can also use pre-built provider, please visit the ["Using Providers and Modules"-Guide](../working-with-cdk-for-terraform/using-providers-and-modules.md) for more on that topic.
+To save some time you can also use pre-built provider, please visit the ["Using Providers"-Guide](../working-with-cdk-for-terraform/using-providers.md) for more on that topic.
```
Comment (ansgarm) [inline docs/getting-started/java.md:156]: *providers

File: docs/getting-started/go.md [hunk 9]
```diff
@@ -158,7 +158,7 @@ You specify it in your local `cdktf.json` file, like this:
 ```
 
 Next you need to run `cdktf get` which generates provider-specific bindings.
-To save some time you can also use pre-built provider, please visit the ["Using Providers and Modules"-Guide](../working-with-cdk-for-terraform/using-providers-and-modules.md) for more on that topic.
+To save some time you can also use pre-built provider, please visit the ["Using Providers"-Guide](../working-with-cdk-for-terraform/using-providers.md) for more on that topic.
```
Comment (ansgarm) [inline docs/getting-started/go.md:161]: *providers

File: docs/getting-started/csharp.md [hunk 10]
```diff
@@ -147,7 +147,7 @@ You specify it in your local `cdktf.json` file, like this:
 ```
 
 Next you need to run `cdktf get` which generates provider-specific bindings.
-To save some time you can also use pre-built provider, please visit the ["Using Providers and Modules"-Guide](../working-with-cdk-for-terraform/using-providers-and-modules.md) for more on that topic.
+To save some time you can also use pre-built provider, please visit the ["Using Providers"-Guide](../working-with-cdk-for-terraform/using-providers.md) for more on that topic.
```
Comment (ansgarm) [inline docs/getting-started/csharp.md:150]: *providers

File: README.md [hunk 11]
```diff
@@ -100,7 +100,7 @@ Choose a language:
 ## Documentation
 
 - Install and run a quick start tutorial at [HashiCorp Learn](https://learn.hashicorp.com/terraform/cdktf/cdktf-install)
-- Learn how to use any [module or provider](./docs/working-with-cdk-for-terraform/using-providers-and-modules.md) with the CDK for Terraform. You can find all modules and providers in the [Terraform Registry](https://registry.terraform.io/).
+- Learn how to use any [module](./docs/working-with-cdk-for-terraform/using-modules.md) and [provider](./docs/working-with-cdk-for-terraform/using-providers.md) with the CDK for Terraform. You can find all modules and providers in the [Terraform Registry](https://registry.terraform.io/).
```
Comment (ansgarm) [inline README.md:103]: any *Terraform module

File: docs/working-with-cdk-for-terraform/using-modules.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+
+To get typed module bindings you first need to add the module in your `cdktf.json` file. For a local module it might look like this:
+
+```jsonc
+{
+  // ...
+  "terraformModules": [
+    {
+      "name": "my-local-module",
+      "source": "./path/to/local/terraform/module"
```
Comment (ansgarm) [inline docs/working-with-cdk-for-terraform/using-modules.md:18]: relative to which directory? Might be useful to state that

File: docs/working-with-cdk-for-terraform/using-modules.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+});
+
+new TestResource(stack, "resource", {
+  name: module.getString("name"),
+});
+```
+
+## Working with Module Outputs
+
+### Modules with bindings
```
Comment (ansgarm) [inline docs/working-with-cdk-for-terraform/using-modules.md:53]: *generated bindings

**Review by ansgarm — CHANGES_REQUESTED** [review]

**Review by ansgarm — APPROVED** [review]

**Pull request merged**